Update Controllers to version 33.0.0

tommasini opened the pull request:
**Description**
Updating the controllers to version 33.0.0, will update the way Network Controller works
It was updated the PhishingControler
The CollectiblesController was renamed to NftController

**Test Cases**
Case1:
* Tested swaps on mainnet (uniswap) and custom network (Binance Smart Chain) (pancakeswap)
Case2:
* Tested transfer of collectible to myself on Mainnet
Case3:
* Tested token detection and nft Detection
Case4:
* Tested transfers between accounts, and the gas fee seems to be working and well defined (confirmed with etherscan and polyscan)
Case 5:
* Tested deep link with network changing
Case 6:
* Tested the activity view in different networks
Case 7:
* Tested switching accounts and verifying the balance on them (Also switching networks)
Case 8:
* Tested transfer collectables to other account on goerly
Case 9:
* Tested to create an new account it should change to it 


Would be good to do an overall test of the app.

**Issue**

Progresses #5092 

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

sethkfman committed:
defined new network controller and imported latest controllers

sethkfman committed:
updated callback functions for controller depending on network state change

sethkfman committed:
troubleshooting callbacks

sethkfman committed:
merged main and updated onNetworkStateChange to use controllemessenger

tommasini committed:
unsubscribe of controller messenger on disconnect working

tommasini committed:
remove logs

tommasini committed:
remove useless refresh from the engine constructor

sethkfman commented on app/core/BackgroundBridge/BackgroundBridge.js:
```diff
@@ -89,7 +89,10 @@ export class BackgroundBridge extends EventEmitter {
 
-    Engine.context.NetworkController.subscribe(this.sendStateUpdate);
+    Engine.controllerMessenger.subscribe(
+      `NetworkController:stateChange`,
```
Maybe we can create a const and reference throughout the codebase. I see this in a few locations

tommasini replied:
Completely agree 💯

tommasini committed:
moved repeated string to a app constant

tommasini committed:
Merge branch 'main' into update/network-contoller

sethkfman commented on app/core/BackgroundBridge/BackgroundBridge.js:
```diff
@@ -89,7 +89,10 @@ export class BackgroundBridge extends EventEmitter {
 
-    Engine.context.NetworkController.subscribe(this.sendStateUpdate);
+    Engine.controllerMessenger.subscribe(
+      `NetworkController:stateChange`,
```
Missed one

sethkfman commented on app/core/BackgroundBridge/BackgroundBridge.js:
```diff
@@ -266,7 +269,10 @@ export class BackgroundBridge extends EventEmitter {
     this.disconnected = true;
-    Engine.context.NetworkController.unsubscribe(this.sendStateUpdate);
+    Engine.controllerMessenger.unsubscribe(
+      `NetworkController:stateChange`,
```
Missed one

sethkfman commented on app/core/EngineService.ts:
```diff
@@ -65,8 +65,15 @@ class EngineService {
+          : Engine.controllerMessenger.subscribe(key, update_bg_state_cb);
+      else
+        Engine.controllerMessenger.subscribe(
+          'NetworkController:stateChange',
```
Missed one

tommasini committed:
replace to const

Gudahtt commented:
I have created a PR to account for the PhishingController breaking changes included in the v31 and v32 controllers releases: #5215

Gudahtt commented on app/core/Engine.js:
```diff
@@ -84,61 +85,77 @@ class Engine {
+        state: initialState.networkController,
+        messenger: this.controllerMessenger.getRestricted({
+          name: 'NetworkController',
+          allowedEvents: [
+            AppConstants.NETWORK_STATE_CHANGE_EVENT,
+            'NetworkController:providerChange',
+          ],
```
Nit: I don't think any allowed events are needed here. It should automatically have permission to use any events or actions under its own namespace

```suggestion
          allowedEvents: [],
```

tommasini replied:
Confirmed

Gudahtt commented on app/core/Engine.js:
```diff
@@ -378,14 +416,14 @@ class Engine {
       } = this.context;
 
       collectibles.setApiKey(process.env.MM_OPENSEA_KEY);
-      network.refreshNetwork();
```
Interesting - even in the older version of the controllers package, this was a private method. It seems that setting `providerConfig` has basically the same effect as calling this, so I agree that it is not needed.

Gudahtt reviewed (commented):
The NetworkController changes all look good to me!

tommasini committed:
Update app/core/Engine.js

Co-authored-by: Mark Stacey <markjstacey@gmail.com>

Gudahtt committed:
Update controllers to v33 (#5216)

* Update controllers to v33

The controllers package has been updated to v33. This release included
various breaking changes relating to NFTs. All controller APIs and
state have been updated to use the term "NFT" instead of "collectible".

* Fix mistake in test fixture

sethkfman commented:
@SocketSecurity ignore @ethereumjs/rlp@4.0.0 rlp@2.2.6 rlp@3.0.0

sethkfman reviewed (approved):
LGTM

Gudahtt commented:
This PR still needs to be merged into this: https://github.com/MetaMask/metamask-mobile/pull/5215

Otherwise the phishing configuration will never be updated

Gudahtt committed:
Handle PhishingController breaking changes (#5215)

* Handle PhishingController breaking changes

The PhishingController was changed in v31 and v32 to no longer update
the phishing configuration regularly in the background. Instead it has
to be updated manually.

We now check for phishing configuration updates each time we're using
it to test whether a URL is suspicious or not. This matches how the
extension works. It should be roughly equivalent to how the Phishing
Controller worked in past versions, except that now it will no longer
be polling for updates when the browser is not in-use.

The phishing configuration is also explicitly updated once upon
construction.

* Fix PhishingController reference

Co-authored-by: tommasini <46944231+tommasini@users.noreply.github.com>

Co-authored-by: tommasini <46944231+tommasini@users.noreply.github.com>

Gudahtt commented on app/store/migrations.js:
```diff
@@ -300,6 +300,32 @@ export const migrations = {
+  },
 };
 
 export const version = 11;
```
Oops, I think I missed this!
```suggestion
export const version = 12;
```

tommasini replied:
Oops, escaped me as well! Thanks for backing here!

tommasini committed:
Update app/store/migrations.js

Co-authored-by: Mark Stacey <markjstacey@gmail.com>

tommasini committed:
new yarn.lock

tommasini committed:
Merge branch main into update/network-controller

tommasini committed:
Merge branch 'main' into update/network-contoller

Gudahtt commented:
I'm seeing this warning during the `setup` step in CI:

```
Warning: patch-package detected a patch file version mismatch

  Don't worry! This is probably fine. The patch was still applied
  successfully. Here's the deets:

  Patch file created for

    @metamask/controllers@30.3.0

  applied to

    @metamask/controllers@33.0.0
  
  At path
  
    node_modules/@metamask/controllers

  This warning is just to give you a heads-up. There is a small chance of
  breakage even though the patch was applied successfully. Make sure the package
  still behaves like you expect (you wrote tests, right?) and then run

    patch-package @metamask/controllers

  to update the version in the patch file name and make this warning go away.

```

Seems like that patch file needs an update

tommasini committed:
update patch for auto swtich accounts

metamaskbot committed:
5.12.0

cortisiko committed:
E2e/merge new approval screen branch (#5320)

* add e2e coverage for automatic security update screen (#5291)

* add e2e coverage for automatic security update screen

* removed unncessary comment.

* fixing contract nickname test

* readd test

* fix test step

* blah.

dan437 committed:
Update @metamask/swaps-controller (#5331)

* @metamask/swaps-controller": "^6.7.1"

* "@metamask/swaps-controller": "^6.7.2"

owencraston committed:
only fetch minimum versions if permissions enabled (#5333)

blackdevelopa committed:
fix 507 (#5337)

wachunei committed:
On-ramp: Add useRegions hook and fix availablePaymentMethods (#5335)

tommasini committed:
WIP: b79dad079 Add strings to feature "Screenshot Warning" (#5289)

sethkfman committed:
version bump for RC 2 (#5344)

metamaskbot committed:
5.12.1

sethkfman committed:
[IMPROVEMENT] Remove Metrics Properties (#5345)

* Remove RPC URL from "Network Switched" event

* Remove RPC URL from "Network Added" event

* Remove block_explorer_url parameter

* removed network_name from all analytics events

* updated snapshot

Co-authored-by: Gustavo Antunes <17601467+gantunesr@users.noreply.github.com>

tommasini committed:
Sanitize privacy settings before sending to Sentry (#5355)

* pairity with extension and removing the device name and the device timezone

* optout metametrics

* mix panel opt out

* removed unnecessary comment

sethkfman committed:
bumped version, added changelog, address audit:ci

Fatxx committed:
Fix onboarding wizard automatic update modal (#5360)

* Fix onboarding tutorial

* Add specific prop to check for EnableAutomaticSecurityChecksModal state

* Update tests

tommasini committed:
missing verification (#5365)

Cal-L committed:
Update copy for Opt in metrics screen and enable custom mainnet RPC (#5362)

* add e2e coverage for automatic security update screen (#5291)

* add e2e coverage for automatic security update screen

* removed unncessary comment.

* fixing contract nickname test

* readd test

* fix test step

* Add network screen to onboarding stack

* Update metametrics screen layout

* Update metametrics copy

* Remove old privacy policy copies

* Begin handling isRPCUpdate prop in NetworkSettings

* Create editable default mainnet

* Fix RPC comparison

* Update ETH custom network name

* Enforce scroll to end on metametrics screen

* e2e changes for metametrics screen

* Fix rpc type warning and invalid URL crasher

* Fix updating network instead of duplicating

* Fix duplicate rpc networks

* Begin consolidating mainnet checks

* Revert "Remove old privacy policy copies"

This reverts commit 029c5b6c076bcdc737db7e1a7496f1284d78f5e1.

* Revert "Update metametrics copy"

This reverts commit 4bb0bd826ba7aa102699803ac4073a1a1a5bd001.

* Revert "Enforce scroll to end on metametrics screen"

This reverts commit 8606ec71f1ab1c5daf3bd14bed576288040a4f93.

* Revert "Update metametrics screen layout"

This reverts commit 13cfe0d3bd3bc3bc533d88e745441b50f3126482.

* Revert opt in to use old UI. Update copy. Hide NFT banner for non default mainnet

* Fix privacy policy copy

* Remove send changes for another PR

* Check for token contract address by chainId

* add swipe up

* Fix unit test and lint error

Co-authored-by: Curtis David <Curtis.David7@gmail.com>
Co-authored-by: CW <chris.wilcox@consensys.net>

wachunei committed:
On-ramp Refactor wyre authentication url approach (#5366)

sethkfman committed:
version bump & changelog update (#5372)

chrisleewilcox commented:
Swaps:
Polygon
  MATIC => WMATIC => Edit gas fee to Market: PASS
  MATIC => UNI: PASS
  USDT => LINK: PASS
  UNI => DAI: PASS
  Cancel transaction: having problems cancelling.  Getting console warning on cancel: https://recordit.co/9xDj8XntwW

chrisleewilcox commented:
NFTs

Autodetect NFTs:
- ETH Mainnet: PASS

chrisleewilcox commented:
Wallet view:

Token fiat balance for asset is incorrect.  See [here](![image](https://user-images.githubusercontent.com/6626407/207095098-bd35c373-76d4-4e47-8d19-dfdf44388275.png)).

chrisleewilcox reviewed (changes requested):
Please review comments.

chrisleewilcox commented:
Send Tokens:

ETH QRC:
ETH Deeplink:
Polygon QRC:
Polygon Deeplink:
BNB QRC:
BNB Deeplink:
AVAX QRC:
AVAX Deeplink:
Arbitrum QRC:
Arbitrum Deeplink:
Optimism QRC:
Optimism Deeplink:

Gudahtt commented:
Are these errors reproducible on `main`? Also, when these errors were encountered, did you recently switch networks?

It's hard to see how these changes would cause cancellation failures, edit limit failures, balance failures, etc. unless maybe there's a bug related to network switching. The code changes barely touch the transaction controller or asset management parts at all.

chrisleewilcox commented:
dApps

Uniswap:

- swaps:

Sushiswap swap:

- swaps:

pancake swap:

- swaps:

1inch:

- swaps:

sethkfman committed:
Merge branch 'main' into release/5.12.1

tommasini committed:
Merge branch 'main' into update/network-contoller

tommasini committed:
Merge branch 'release/5.12.1' into update/network-contoller

tommasini committed:
Merge branch 'main' into update/network-contoller

chrisleewilcox reviewed (approved):
LGTM
